src/lib/y2storage/proposal_settings.rb

```diff
@@ -84,32 +137,74 @@ class ProposalUserSettings
+    # @note :legacy and :ng formats
     # @return [Symbol] what to do regarding removal of existing Linux
     #   partitions. See {DiskAnalyzer} for the definition of "Linux partitions".
     #   @see #windows_delete_mode for the possible values and exceptions
-    attr_accessor :linux_delete_mode
+    attr_reader :linux_delete_mode
 
+    # @note :legacy and :ng formats
     # @return [Symbol] what to do regarding removal of existing partitions that
     #   don't fit in #windows_delete_mode or #linux_delete_mode.
     #   @see #windows_delete_mode for the possible values and exceptions
-    attr_accessor :other_delete_mode
-
-    def initialize
-      @use_lvm                  = false
-      self.encryption_password  = nil
-      @root_filesystem_type     = Filesystems::Type::BTRFS
-      @use_snapshots            = true
-      @use_separate_home        = true
-      @home_filesystem_type     = Filesystems::Type::XFS
-      @enlarge_swap_for_suspend = false
-      @resize_windows           = true
-      @windows_delete_mode      = :ondemand
-      @linux_delete_mode        = :ondemand
-      @other_delete_mode        = :ondemand
+    attr_reader :other_delete_mode
+
+    # @note :ng format
+    # @return [Symbol] if the user decides to use LVM, strategy to decide the size
+    #   of the volume group (and, thus, the number and size of created physical volumes).
+    #
+    #   * :use_available The VG will be created to use all the available space, thus the
+    #     VG size could be greater than the sum of LVs sizes.
+    #   * use_needed: The created VG will match the requirements 1:1, so its size will be
+    #     exactly the sum of all the LVs sizes.
+    #   * use_vg_size: The VG will have a predefined size, that could be greater than the
+    #     LVs sizes.
+    attr_reader   :lvm_vg_strategy
+
+    # @note :ng format
+    # @return [DiskSize] if :use_vg_size is specified in the previous option, this will
+    #   specify the predefined size of the LVM volume group.
+    attr_accessor :lvm_vg_size
+
+    # @note :ng format
+    # @return [Array<VolumeSpecification>] list of volumes specifications used during
+    #   the proposal
+    attr_accessor :volumes
+
+    alias_method :lvm, :use_lvm
+    alias_method :lvm=, :use_lvm=
+
+    LEGACY_FORMAT = :legacy
+    NG_FORMAT = :ng
+
+    # New object initialized according to the YaST product features (i.e. /control.xml)
+    # @return [ProposalSettings]
+    def self.new_for_current_product
+      settings = new
+      settings.for_current_product
+      settings
+    end
+
+    # Set settings according to the current product
+    def for_current_product
+      apply_defaults
+      load_features
+    end
+
+    # Current format used in control file
+    #
+    # Format is considered :ng only if subsections 'proposal' and 'volumes' are
+    # present in the 'partitioning' section.
+    #
+    # @return [Symbol] :ng or :legacy
+    def format
+      ng_format? ? NG_FORMAT : LEGACY_FORMAT
```

Comment:
I see a problem with the design of `#format`, `#ng_format?` and so on. `ProductFeatures` is not immutable. Quite the opposite, nowadays looks like we are introducing more and more ways of changing it. So the mode should not be determined dynamically based on its current content. It should be determined during `#load_features` and stored in the `ProductFeatures` object alongside the read features.

Reply:
Stored in the `ProductFeatures`? Do you mean `ProposalSettings`?